### PR TITLE
Employee Management Page - Data Retrieval

### PR DESCRIPTION
**Pull request by osaosula**
As A User
I want to view a list of Employees on the Employee Management Page that were retrieved from the “/employees” endpoint

So that data from the endpoint is viewable instead of hard coded data

 
Using Tanstack Query - request all employees from the “/employee” endpoint, ensuring that the loading and error states and messages are returned. The handling of errors is out of scope

While data is being retrieved, display a loading message “loading…” The AG-Grid should not be visible until the data has been returned from the endpoint. (You can check the “data” property of the response. If the request was successful, the “data” property will contain the list of employees)

### CODE REVIEW

File: src/pages/Employees.jsx [hunk 1]
```diff
@@ -1,77 +1,20 @@
...
+  const {
+    isPending,
+    data: employees,
+    error,
+  } = useQuery({
+    queryKey: ["employees"],
+    queryFn: getEployees,
+  });
+
+  const [rowData] = [employees];
```
Comment (BorisLaboratory) [inline src/pages/Employees.jsx:18]: empty line . needs to be closed

File: src/pages/Employees.jsx [hunk 2]
```diff
@@ -1,77 +1,20 @@
...
-  ]);
+  const {
+    isPending,
+    data: employees,
+    error,
+  } = useQuery({
+    queryKey: ["employees"],
+    queryFn: getEployees,
+  });
+
```
Comment (BorisLaboratory) [inline src/pages/Employees.jsx:16]: empty line . needs to be closed

**Review by BorisLaboratory — CHANGES_REQUESTED** [review]
Looks good. However quite a number of empty lines were observed . see the comments

File: src/services/apiEmployees.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+export async function getEployees() {
+  const { data, error } = await axios.get(
+    "https://v5bgbb7bq3.execute-api.us-east-1.amazonaws.com/development/employees"
```
Comment (BorisLaboratory) [inline src/services/apiEmployees.js:4]: Hard-coded URL can be stored in a constant

**Review by BorisLaboratory — CHANGES_REQUESTED** [review]

File: src/services/apiEmployees.js (hunk 3 above)
Comment (osaosula) [reply src/services/apiEmployees.js:4]: Fixed

File: src/pages/Employees.jsx [hunk 4]
```diff
@@ -93,6 +36,9 @@ const Employees = () => {
     { field: "lastLogin", headerName: "Last Login", flex: 1 },
   ];
```
[the remark this answers is not in the document]
Comment (osaosula) [reply src/pages/Employees.jsx:38]: Fixed

File: src/pages/Employees.jsx [hunk 5]
```diff
@@ -93,6 +36,9 @@ const Employees = () => {
...
 
+  if (isPending) return "loading...";
+  if (!employees) return "No Employee records";
+
```
[the remark this answers is not in the document]
Comment (osaosula) [reply src/pages/Employees.jsx:41]: Fixed

File: src/pages/Employees.jsx (hunk 1 above)
Comment (osaosula) [reply src/pages/Employees.jsx:18]: Fixed

File: src/pages/Employees.jsx (hunk 2 above)
Comment (osaosula) [reply src/pages/Employees.jsx:16]: Fixed

File: src/App.jsx [hunk 6]
```diff
@@ -8,56 +8,69 @@ import Layout from "./pages/Layout";
 import { AuthProvider } from "./hooks/useAuth";
 import { ProtectedRoute } from "./components/ProtectedRoute";
 import { Routes, Route } from "react-router-dom";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+
```
[the remark this answers is not in the document]
Comment (osaosula) [reply src/App.jsx:13]: Fixed

**Review by BorisLaboratory — DISMISSED** [review]
Looks good

**Pull request closed without merging**